### PR TITLE
releng: promote kube-cross/go-runner/releng-ci Go1.17.6 and 1.16.13 images

### PR DESCRIPTION
**Pull request by palnabarun**
### What type of PR is this?

Used `cip-mm` to create the changes.

/kind feature
/area dependency

### What this PR does / why we need it:

Tracking issue: [kubernetes/release#2349](https://github.com/kubernetes/release/issues/2379)

  - kube-cross: v1.21.0-go1.16.13-buster.0 / v1.22.0-go1.16.13-buster.0 / v1.23.0-go1.17.6-bullseye.0 / v1.24.0-go1.17.6-bullseye.0
  - go-runner: v2.3.1-go1.16.13-buster.0 / v2.3.1-go1.17.6-bullseye.0
  - releng-ci: 1.16.13-buster-0 / 1.17.6-bullseye-0

/assign @justaugustus @saschagrunert @puerco @xmudrii @Verolop
cc @kubernetes/release-engineering

Signed-off-by: Nabarun Pal <pal.nabarun95@gmail.com>

### CODE REVIEW

File: k8s.gcr.io/images/k8s-staging-build-image/images.yaml [hunk 1]
```diff
@@ -384,9 +385,30 @@
     "sha256:d2165f43652fcca3e480eea3bb1f267d862bf39ef30f79bd6460e4230589b89d": ["v2.3.1-go1.16.12-buster.0"]
     "sha256:d3d705c7de100a7cf49df13c4198d156078d6273032de503f723ba1534a6170f": ["v2.3.1-go1.15.12-buster.0"]
     "sha256:dafae2d33f841f6c19381ee2a680df6f65ffee6df9e99e4b88f7e38991afc72c": ["v2.3.1-go1.17.2-buster.0"]
+    "sha256:e9c0cb5a75e21858f1dd7e5d15cc149cb88833d00bb18100886f536826ef8561": ["v2.3.1-go1.17.6-bullseye.0"]
     "sha256:ebca4d7c6f2b4250b2d79d5310ee851d0532a302517e81348aa783cfe58024e4": ["v2.3.1-go1.17.5-bullseye.0"]
     "sha256:fda5b13760fd7588c5cd0be8c58b9621463aa08cab7fb3e38f01d7a314e8d7f7": ["v2.3.1-go1.15.10-buster.0"]
     "sha256:ff6e2f3683e7d284674ed18341fc898060204e8c43c9b477e13c6f7faf3e66d4": ["buster-v2.0.0"]
+- name: go-runner-amd64
```
Comment (cpanato) [inline k8s.gcr.io/images/k8s-staging-build-image/images.yaml:392]: those images we don't promote

File: k8s.gcr.io/images/k8s-staging-build-image/images.yaml [hunk 2]
```diff
@@ -484,6 +510,24 @@
     "sha256:ef3d86412d9335e7d587aebef9bc004bd42845516fee11e830a49d5e55aaaad0": ["v1.15.10-legacy-1"]
     "sha256:f393f7b488a9488fc1dff42ba24e2674f9a2d962c7dedd93a457aaf414ab956e": ["v1.15.15-legacy-1"]
     "sha256:f9f569e6e1da0fa6ac441b625e599bead8ab10f480ad2a84966cd4a0091e76a8": ["v1.12.17-2"]
+- name: kube-cross-amd64
```
Comment (cpanato) [inline k8s.gcr.io/images/k8s-staging-build-image/images.yaml:513]: same for these ones

File: k8s.gcr.io/images/k8s-staging-build-image/images.yaml (hunk 1 above)
Comment (cpanato) [reply k8s.gcr.io/images/k8s-staging-build-image/images.yaml:392]: PR to update: https://github.com/kubernetes/k8s.io/pull/3271

Comment (palnabarun) [reply k8s.gcr.io/images/k8s-staging-build-image/images.yaml:392]: Ack!

Just curious, why don't we promote them?

Comment (cpanato) [reply k8s.gcr.io/images/k8s-staging-build-image/images.yaml:392]: they are already part of the image

```
crane manifest k8s.gcr.io/build-image/go-runner:v2.3.1-go1.17.6-bullseye.0
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 737,
         "digest": "sha256:6078e5b3ef3fd90d8b7813a01064d736fd8367bdabd72eabfb76ddf7972ba844",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 737,
         "digest": "sha256:40479777cbfedd23a93f5638d0b2af7e2f58cdd1ccdbcd0655878ae46612a82b",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v7"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 737,
         "digest": "sha256:73369df8858451f414cf636259ea7a1b496c81bae6cd58da08c7d410aec8febe",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 737,
         "digest": "sha256:0cb0e1ab2039fddb87134583ca600d5a13c8fcda7857e1d8e9f95397f8f360b7",
         "platform": {
            "architecture": "ppc64le",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 737,
         "digest": "sha256:b8e0416f567f12e1490bbf0409f17a9b315b37189af3c09c2a39870c49e67a30",
         "platform": {
            "architecture": "s390x",
            "os": "linux"
         }
      }
   ]
}
```

Comment (palnabarun) [reply k8s.gcr.io/images/k8s-staging-build-image/images.yaml:392]: Interesting! Thanks for the context.

Shall we make changes in the manifest grower to ignore the platform variants if the parent manifest has them already?

Comment (cpanato) [reply k8s.gcr.io/images/k8s-staging-build-image/images.yaml:392]: sounds like a good idea, can you open an issue? i think the krel command was removing those, you can take a look at it as well

Comment (palnabarun) [reply k8s.gcr.io/images/k8s-staging-build-image/images.yaml:392]: Created an issue in k-sigs/promo-tools: https://github.com/kubernetes-sigs/promo-tools/issues/484

I planned to use `kpromo pr` but at this point in time it doesn't take `--staging_repo` like argument.

**Pull request merged**